src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -631,18 +632,24 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                 Debug.Assert(((VariableDeclarationSyntax)topLevelVariableDeclaration).Deconstruction.Value != null);
 
                 var statement = topLevelVariableDeclaration.Parent;
+                TypeSymbol result;
                 switch (statement.Kind())
                 {
                     case SyntaxKind.LocalDeclarationStatement:
                         this.binder.BindLocalDeclarationStatement((LocalDeclarationStatementSyntax)statement, diagnostics);
+                        break;
 
-                        TypeSymbol result = this._type;
-                        Debug.Assert((object)result != null);
-                        return result;
+                    case SyntaxKind.ForStatement:
+                        this.binder.BindFor((ForStatementSyntax)statement, diagnostics);
```

Comment:
Are we binding the `for` twice? Once here and once when binding the method body statements? If so, it seems unnecessary to bind anything other than the declarations here. Otherwise we're doing extra work and potentially reporting extra diagnostics.